packages/cosmic-swingset/lib/launch-chain.js

```diff
@@ -135,6 +139,8 @@ export async function launch(
     const blockStart = now;
     let stepsRemaining = FIXME_MAX_CRANKS_PER_BLOCK;
     let stepped = true;
+    // TODO: how to rewrite without an await in loop?
```

Comment:
maybe we can add a kernel API to run up-to N cranks, and use the slog hooks you added to capture the runtime of each

OTOH this will be replaced by the escalators sooner or later, so if we're comfortable with the `await` style violation for a while, it will go away by itself

Reply:
> maybe we can add a kernel API to run up-to N cranks, and use the slog hooks you added to capture the runtime of each

Maybe add an argument to `controller.run()`, a la `run(maxSteps = Infinity)`?

That slog hooks sounds like a good approach, and we should use them to feed into the OpenTelemetry metrics instead of a separate step-one-at-a-time mechanism.

The bootstrap and normal cases should use the same mechanics.as well.

> OTOH this will be replaced by the escalators sooner or later, so if we're comfortable with the `await` style violation for a while, it will go away by itself

Duly noted.